[SW-265] Spot Ros2 slight refactor for testability and unit testing

mpickett-bdai opened the pull request:
Tickets: SW-265, SW-255
Changes:
- Made Spot Ros 2 "Mock_spot" actually work as intended allowing services to be created (they already had checks for mock)
- Added unit tests and changes to spot_ros2.py to allow for that testing without a launch file
- To fix dependency issues, __init__.py needed moving into the tests folder so that it could properly find modules
- Cleaned up redundant code of the camera publishing using a partial and the common naming scheme (this does potentially change the topic name on the rgb camera images. Its cleaner if the spot_wrapper image name and the topic name match, but if it's a deal breaker I'm happy to put a check in to swap just that one to be "camera/..." instead of "visual/.." again)

Tests run:
- Ran on a spot testing out normal functionality as well as tested the unit tests
- Viewed images in Rviz2 to make sure they were publishing

amessing-bdai commented on spot_driver/spot_driver/spot_ros2.py:
```diff
@@ -908,55 +873,43 @@ def publish_graph_nav_pose_callback(self) -> None:
+            setattr(
+                self,
+                f"{camera_name}_{image_type}_pub",
+                self.create_publisher(Image, f"{image_type}/{camera_name}/image", 1),
```
I am thinking keep the public interface the same - otherwise its a breaking change

mpickett-bdai replied:
I had a feeling that might be the case, but thought I'd point out the inconsistency. I'll through up a small patch to set it to "camera"

mpickett-bdai commented:
@ksharma-bdai I added you just because you had a camera related PR that was merged in while this was still in review. I believe your changes are unaffected and I fixed any conflicts, but thought I'd have you take a quick look at the publish side at least.

amessing-bdai commented:
@mpickett-bdai Is this good to go?

mpickett-bdai commented:
> @mpickett-bdai Is this good to go?

I believe so, updating the branch now then I can merge.

mpickett-bdai commented:
> > @mpickett-bdai Is this good to go?
> 
> I believe so, updating the branch now then I can merge.

Nevermind I'm not authorized to merge, but yeah looks good to go.
